Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1368,6 +1906,31 @@ void TextureCacheBase::CopyRenderTargetToTexture(u32 dstAddr, EFBCopyFormat dstF
   }
 }
 
+void TextureCacheBase::SetXFBMemoryToKeyColor(u8* dst, u32 stride, u32 bytes_per_row,
```

Comment:
Wouldn't it be better to do this in the shader? Seems like a heavy loop to do on the CPU every copy.

Reply:
This matches what we do in EFB where we clear out the memory to 0.  Not disagreeing with you but I imagine both should be handled at the same time.  Probably outside of this PR.

Reply:
A memset call is much cheaper than testing every texel and branching. I think it's probably a worthwhile optimization, and not complex.

Reply:
Agreed, I used a memset originally but it wasn't working.  That's what phire's original implementation did but it didn't work when I implemented.  Not sure what I did wrong.  I can take another pass at it.

Reply:
As far as I can tell, it should be just a matter of comparing the RGB colour against the fixed range, and if it exceeds it, emitting the "key" instead. Although, another thing to consider; how does the hardware behave here? Writing a fixed YUV colour seems unlikely.

Reply:
> As far as I can tell, it should be just a matter of comparing the RGB colour against the fixed range, and if it exceeds it, emitting the "key" instead.

This is a case where we don't have a value in memory though.  So we have to write something.  We could write 0s like is done on EFBs but that causes issues since some games use values outside the yuv range to mean something.  This was all done by phire originally and you can read his comments about it in his very first PR 2051.  I will say, one of the games this was meant to fix "Mario Strikers" it doesn't help in XFB2Tex since the background texture is invalidated when this XFB with transparency (that is supposed to overlay on top) appears.  However, XFB2Ram does work properly with this fix since we can just lookup that invalidated texture from memory.

It also does fix "Chicken Little" but likely what you've suggested about outputting black would suffice in that case.

> Although, another thing to consider; how does the hardware behave here? Writing a fixed YUV colour seems unlikely.

Not entirely sure what hardware does.  But it's likely handling partial XFBs somehow.  This implementation was straight from phire's original PR.

Reply:
Ah, I see, this is only for the XFB2Tex path. Sorry for the confusion. In this case, the shader suggestion doesn't make sense then ;)